src/test/java/org/example/bookshop/controller/BookControllerTest.java

```diff
@@ -0,0 +1,230 @@
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+
+        EqualsBuilder.reflectionEquals(bookDto, actual);
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:database/books/add-two-books-to-books-table.sql",
+            "classpath:database/categories/add-two-category-to-books_categories-table.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/books/delete-two-books-from-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @WithMockUser(roles = {"USER"})
+    @DisplayName("Get all books")
+    void getAllBooks_Success() throws Exception {
+        BookDto secondBookDto = new BookDto();
+        secondBookDto.setId(SECOND_ID);
+        secondBookDto.setTitle(BOOK_TITLE + UNIQUE_PARAM);
+        secondBookDto.setAuthor(BOOK_AUTHOR + UNIQUE_PARAM);
+        secondBookDto.setIsbn(BOOK_ISBN + UNIQUE_PARAM);
+        secondBookDto.setPrice(BOOK_PRICE);
+        secondBookDto.setCategoryIds(categoriesId);
+        BookDto thirdBookDto = new BookDto();
+        thirdBookDto.setId(THIRD_ID);
+        thirdBookDto.setTitle(UNIQUE_PARAM + BOOK_TITLE);
+        thirdBookDto.setAuthor(UNIQUE_PARAM + BOOK_AUTHOR);
+        thirdBookDto.setIsbn(UNIQUE_PARAM + BOOK_ISBN);
+        thirdBookDto.setPrice(BOOK_PRICE);
+        thirdBookDto.setCategoryIds(categoriesId);
+        List<BookDto> expected = List.of(bookDto, secondBookDto, thirdBookDto);
+
+        MvcResult result = mockMvc.perform(
+                get("/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(
+                result.getResponse().getContentAsByteArray(), BookDto[].class);
+        List<BookDto> actualList = Arrays.stream(actual).toList();
+        Assertions.assertEquals(expected.size(), actualList.size());
+        Assertions.assertTrue(IntStream.range(0, expected.size())
+                .allMatch(i -> EqualsBuilder.reflectionEquals(expected.get(i), actualList.get(i))));
+    }
+
+    @Test
+    @WithMockUser(roles = {"USER"})
+    @DisplayName("Get book by id")
+    void getBookById_ValidId_Success() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/books/{id}", FIRST_ID))
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+
+        EqualsBuilder.reflectionEquals(bookDto, actual);
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:database/books/delete-all-books-from-books-table.sql",
+            "classpath:database/books/add-book-to-books-table.sql",
+            "classpath:database/categories/add-category-for-books_categories-table.sql"},
```

Comment:
Why do you use the add script after the test method? 
Also if you execute the same scripts for each test method you can move these scripts above the class